File: Two-Sum.py
#https://leetcode.com/problems/two-sum/description/
#Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
class Solution:
   def twoSum(self, nums: list[int], target: int) -> list[int]:
       for i in range (0,len(nums)):
           x=nums[i]
           y=target-x
           for j in range(0,len(nums)):
               if nums[j]==y:
                   if i==j:
                       j+=1
                   else:
                       out=[]
                       out.append(i)
                       out.append(j)
                       return out
               else:
                   j+=1
           i+=1

File: TwoSum.java
import java.util.HashMap;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> set = new HashMap<>();
        //we are going to look at the hashmap to find the number which when added to nums[i] gives the target value
        for(int i=0;i<=nums.length;i++){
            int diff=target-nums[i]; //eg target=9, nums[0]=2. 9-2=7
            if(set.containsKey(diff)){//check to see if 7 is in the HashMap before adding the element to it
                return new int[] {set.get(diff),i};//returns the array containing the indices of the pair of numbers
            }
            set.put(nums[i],i);//add the value and index to the hashmap
        }
        return nums;
    }
}
